Tests/PackageGraphPerformanceTests/DependencyResolverPerfTests.swift

```diff
@@ -0,0 +1,154 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+
+import Basic
+import PackageGraph
+
+import struct Utility.Version
+
+import TestSupport
+
+#if ENABLE_PERF_TESTS
+
+private let v1: Version = "1.0.0"
+private let v1Range: VersionSetSpecifier = .range("1.0.0" ..< "2.0.0")
+
+class DependencyResolverPerfTests: XCTestCase {
+
+    func testTrivalResolution_X1000() {
+        let N = 1000
+        // Try resolving a trivial graph:
+        //        ↗ C
+        // A -> B
+        //        ↘ D
+        let provider = MockPackagesProvider(containers: [
+            MockPackageContainer(name: "A", dependenciesByVersion: [
+                v1: [(container: "B", versionRequirement: v1Range)]]),
+            MockPackageContainer(name: "B", dependenciesByVersion: [
+                v1: [
+                    (container: "C", versionRequirement: v1Range),
+                    (container: "D", versionRequirement: v1Range),
+                ]
+            ]),
+            MockPackageContainer(name: "C", dependenciesByVersion: [
+                v1: []]),
+            MockPackageContainer(name: "D", dependenciesByVersion: [
+                v1: []])
+        ])
+        let resolver = MockDependencyResolver(provider, MockResolverDelegate())
+        measure {
+            for _ in 0..<N {
+                let result = try! resolver.resolve(constraints: [MockPackageConstraint(container: "A", versionRequirement: v1Range)])
+                XCTAssertEqual(result, ["A": v1, "B": v1, "C": v1, "D": v1])
+            }
+        }
+    }
+
+    func testResolutionWith100Depth0Breadth_X1() {
+        let N = 1
+        let depth = 100
+        let breadth = 0
+
+        let graph = createDummyGraph(depth: depth, breadth: breadth)
+        let resolver = MockDependencyResolver(MockPackagesProvider(containers: graph.containers), MockResolverDelegate())
+
+        measure {
+            for _ in 0..<N {
+                let result = try! resolver.resolve(constraints: [graph.rootConstraint])
```

Comment:
I think this should also create the resolver inside the loop, so we aren't being effected by caching inside the resolver (which it may eventually do, in which case we would probably want a perf test for both situations).